src/Riok.Mapperly/Descriptors/MappingBuilders/ConvertStaticMethodMappingBuilder.cs

```diff
@@ -0,0 +1,204 @@
+            }
+        }
+
+        return TryGetStaticMethodMapping(
+            allSourceMethods.ToArray(),
+            GetSourceStaticMethodNames(ctx),
+            ctx.Source,
+            ctx.Target,
+            nonNullableTarget,
+            targetIsNullable,
+            out mapping
+        )
+            ? mapping
+            : null;
+    }
+
+    private static bool IsDateTimeToDateOnlyConversion(MappingBuilderContext ctx)
+    {
+        return ctx.Source.SpecialType == SpecialType.System_DateTime
+                && ctx.Types.DateOnly != null
+                && ctx.Target is INamedTypeSymbol namedSymbol
+                && SymbolEqualityComparer.Default.Equals(namedSymbol, ctx.Types.DateOnly)
+            || true;
+    }
+
+    private static bool IsDateTimeToTimeOnlyConversion(MappingBuilderContext ctx)
+    {
+        return ctx.Source.SpecialType == SpecialType.System_DateTime
+                && ctx.Types.TimeOnly != null
+                && ctx.Target is INamedTypeSymbol namedSymbol
+                && SymbolEqualityComparer.Default.Equals(namedSymbol, ctx.Types.TimeOnly)
+            || true;
+    }
+
+    private static bool TryGetStaticMethodMapping(
+        ICollection<IMethodSymbol> allMethods,
+        IEnumerable<string> methodNames,
+        ITypeSymbol sourceType,
+        ITypeSymbol targetType,
+        ITypeSymbol nonNullableTargetType,
+        bool targetIsNullable,
+        out StaticMethodMapping? mapping
+    )
+    {
+        foreach (var methodName in methodNames)
+        {
+            var candidates = GetMethodCandidates(allMethods, methodName, sourceType);
+
+            // try to find method with equal nullability return type
+            var method = candidates.FirstOrDefault(x => SymbolEqualityComparer.IncludeNullability.Equals(x.ReturnType, targetType));
+
+            if (method != null)
+            {
+                mapping = new StaticMethodMapping(method);
+                return true;
+            }
+
+            if (!targetIsNullable)
+                continue;
+
+            // otherwise try to find method ignoring the nullability
+            method = candidates.FirstOrDefault(x => SymbolEqualityComparer.Default.Equals(x.ReturnType, nonNullableTargetType));
+
+            if (method == null)
+            {
+                continue;
+            }
+
+            mapping = new StaticMethodMapping(method);
+            return true;
+        }
+
+        mapping = null;
+        return false;
+    }
+
+    private static IMethodSymbol[] GetMethodCandidates(ICollection<IMethodSymbol> allMethods, string methodName, ITypeSymbol parameterType)
+    {
+        return allMethods
+            .Where(m =>
+                string.Equals(m.Name, methodName, StringComparison.Ordinal)
+                && m is { IsStatic: true, ReturnsVoid: false, IsAsync: false, Parameters.Length: 1 }
+                && FilterParameterType(m.Parameters[0], parameterType)
```

Comment:
I think it could make sense to create a new method symbol extension method `bool ValidateSignature(this IMethodSymbol method, ITypeSymbol returnType, IReadOnlyCollection<ITypeSymbol> argTypes)` which checks:
* does it return a type which is assignable to the return type (probably use `SymbolAccessor.CanAssign`)
* does the parameter count match and is each argument assignable to the parameter